docs/how-to-guides/develop-a-tool/use-cascading-inputs-in-tool.md

```diff
@@ -0,0 +1,136 @@
+    * Use the `@tool` decorator to identify the function as a tool.
+    * When an input should only take on a certain set of fixed values, an Enum class such as `UserType` as shown in the following example can be created.
+    * Within the tool's logic, various inputs may be used depending on the value of the `user_type` input.
+
+```python
+from enum import Enum
+
+from promptflow import tool
+
+
+class UserType(str, Enum):
+    STUDENT = "student"
+    TEACHER = "teacher"
+
+
+@tool
+def my_tool(user_type: Enum, student_id: str = "", teacher_id: str = "") -> str:
+    """This is a dummy function to support cascading inputs.
+
+    :param user_type: user type, student or teacher.
+    :param student_id: student id.
+    :param teacher_id: teacher id.
+    :return: id of the user.
+    If user_type is student, return student_id.
+    If user_type is teacher, return teacher_id.
+    """
+    if user_type == UserType.STUDENT:
+        return student_id
+    elif user_type == UserType.TEACHER:
+        return teacher_id
+    else:
+        raise Exception("Invalid user.")
+```
+
+2. Follow the guide [Create and Use Tool Package](create-and-use-tool-package.md) to generate a tool yaml for your tool. Afterwards, manually update this tool yaml to transition from common inputs to cascading inputs.
+Referring to the [tool_with_cascading_inputs.yaml](https://github.com/microsoft/promptflow/blob/main/examples/tools/tool-package-quickstart/my_tool_package/yamls/tool_with_cascading_inputs.yaml) as an example, You need incorporate two configurations for cascading inputs: `enabled_by` and `enabled_by_value`. The `enabled_by_value` in one input means that this input is enabled and displayed by the value of the input referred to in the `enabled_by` attribute.
+
+```yaml
+my_tool_package.tools.tool_with_cascading_inputs.my_tool:
+  function: my_tool
+  inputs:
+    user_type:
+      type:
+      - string
+      enum:
+        - student
+        - teacher
+    student_id:
+      type:
+      - string
+      # This input is enabled by the input "user_type".
+      enabled_by: user_type
+      # This input is enabled when "user_type" is "student".
+      enabled_by_value: [student]
+    teacher_id:
+      type:
+        - string
+      enabled_by: user_type
+      enabled_by_value: [teacher]
+  module: my_tool_package.tools.tool_with_cascading_inputs
+  name: My Tool with Cascading Inputs
+  description: This is my tool with cascading inputs
+  type: python
+```
+> Note: The `enabled_by_value` in the tool yaml is of list type, implying that a single input can be enabled by multiple values from the dependent input.
+
+## Use your tool from VSCode Extension
+Once you've built and shared your tool, you can use your tool from VSCode Extension according to [Create and Use Tool Package](create-and-use-tool-package.md).
```

Comment:
Once you package and share your tool, you can use it in VS Code per the [tool package guide](https://claude.ai/chat/create-and-use-tool-package.md). We have a [demo flow](https://github.com/microsoft/promptflow/tree/main/examples/tools/use-cases/cascading-inputs-tool-showcase) you can try.